File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s_dict = defaultdict(int)
        t_dict = defaultdict(int)

        if len(s) != len(t):
            return False
        
        for i in range(len(s)):
            s_dict[s[i]] += 1
            t_dict[t[i]] += 1
        
        if s_dict == t_dict:
            return True
        else:
            return False

File: 242-valid-anagram/valid-anagram.go
func isAnagram(s string, t string) bool {
    if len(s) != len(t) {
        return false
    }

    hashmap := make(map[rune]int)

    for _, ch := range s {
        hashmap[ch]++
    }

    for _, ch := range t {
        hashmap[ch]--

        if hashmap[ch] < 0 {
            return false
        }
    }

    return true

}